### ISSUE TITLE
support additional record types

### ISSUE THREAD

**Issue by jluebbe** [nsupdate-info/nsupdate.info#105]
It would be nice allow manual configuration of additional records for each host. Examples would be MX, multiple SRV records, TLSA (for DANE).

**Comment by cclements** [comment]
hi, we would very much like to be able to use SRV records specifically for LDAP authentication.

**Comment by assimilat** [comment]
+1 To more record types.
Would it be possible to be able to add tags where the user could specify custom types? By "custom types" I mean that you could just present a 3 or 4 text input form to the user for the record type, record, host and/or domain which would allow the user to specify TXT records or SRV records or whatever DNS record they want. I think if you went that way with a flexible input form like this:
 http://bootsnipp.com/snippets/featured/dynamic-form-fields-add-amp-remove 
Then nsupdate.info would be THE goto dns management tool for me and my shop and likely anyone that finds out about it.

**Comment by pejakm** [comment]
+1 Also, TXT records are mandatory for stuff like SPF, various web services, domain key suppo...

**Comment by silopolis** [comment]
+1

**Comment by luckymedog** [comment]
I've just setup an xmpp/jabber server, and I need (I believe) the SRV record type to be able to use with my dynamic dns.

**Comment by ThomasWaldmann** [comment]
Yes, some addtl. record types are definitely useful. Now it just needs someone who writes the python and html for it.

**Comment by helloray** [comment]
+1 for TXT records.

**Comment by strarsis** [comment]
@ThomasWaldmann: An API for managing records, notably TXT records, would make Let's Encrypt DNS Validation possible, like using [acme.sh](https://github.com/Neilpang/acme.sh/tree/master/dnsapi).

**Comment by ThomasWaldmann** [comment]
Yes.

Note that you can also do ACME via http, without needing addtl. dns records.

**Comment by strarsis** [comment]
@ThomasWaldmann: Correct, but especially for IPv4 internet connections, often non-standard ports are used for HTTP.

**Comment by zroug** [comment]
One thing that does work with http validation but does not work with dns validation is wildcard certificates. It's not very important to me but I just wanted to note that.

**Comment by misery** [comment]
I really like to see TXT support for acme usage. :-)
Any progress Here?

**Comment by ThomasWaldmann** [comment]
No progress. But I see the use case for LE certs, so feel free to do a PR, if you like.

When implementing that, it would be nice to have it somehow "extensible" for other record type, so please not a TXT-only hack.

**Comment by ioogithub** [comment]
It has been 9 years since this original request was made. I am using nsupdate however I do not have a way to update my LE cert. I do not have access to port 80 or 443 so adding a TXT record is my only option. When users with similar constrains mention this in the LE forums the developers recommend changing to one of the many dynDNS service that supports adding TXT records.

I really like nsupdate,info but HTTPS is essential in 2022. Is there any chance this will ever happen?
